---
title: "DoxyPep_DNA_0723"
author: "Abigail Glascock"
date: "2023-07-12"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)

```


```{r import}

library(tidyverse)
library(rstatix)
library(ggpubr)
library(idseqr)
library(vegan)
library(scales)
library(DESeq2)
library(ggrepel)
library(metagenomeSeq)
library(ALDEx2)

rename <- dplyr::rename
select <- dplyr::select

```

```{r theme}
theme_mine <-function(base_size=16,base_family=""){
  theme_bw(base_size,base_family=base_family) %+replace%
    theme(
      strip.background = element_blank(),
      strip.text.x = element_text(size=16),
      strip.text.y = element_text(size=16),
      axis.text.x = element_text(size=16),
      axis.text.y = element_text(size=16,hjust = 1),
      axis.ticks = element_line(colour = "black"),
      axis.title.x = element_text(size = 16),
      axis.title.y = element_text(size = 16,angle = 90, vjust=2),
      legend.position = "top", legend.text=element_text(size=16),
      panel.background = element_blank(),
      panel.border = element_blank(),
      panel.grid.major = element_blank(),
      panel.grid.minor = element_blank(),
      panel.spacing = unit(1.0,"lines"),
      plot.background = element_blank(),
      plot.margin = unit(c(0.5,0.5,0.5,0.5),"lines"),
      axis.line = element_line(colour = "black")
    )
}
theme_set(theme_mine())

```

# IMPORTATION

```{r set paths and read data}


metadata<-read.csv("../Inputs/final_samples.csv", header=TRUE)

ddsdp = readRDS("../Inputs/Microbiome/DNA/dds_object_doxy_genus.rds")

reports_filterbg_padj <- read.csv("../Inputs/Microbiome/DNA/microbiome_reports.csv", header=TRUE)

reports_filterbg_species_padj <- read.csv("../Inputs/Microbiome/DNA/microbiome_reports_species.csv", header=TRUE)

```

# --------------------
# RESULTS: SUPPLEMENTAL FIGURES


## Supplemental Figure 5A: Microbial mass

```{r microbial mass}
### wrangle data ###

reports_2<-reports_filterbg_padj %>%
  dplyr::filter(category=="bacteria")%>%
  dplyr::filter(!grepl("NTC|NEC|EMPTY",sample_name)) %>%
  pivot_wider(names_from = name, values_from = nt_count, values_fill=0, id_cols = sample_name) 
reports_3<-reports_2%>%mutate(sum_bact_count=rowSums(across(where(is.numeric)))) %>%
  mutate(sample_name=toupper(sample_name))
reports_4<-left_join(reports_3,metadata,by="sample_name")
reports_5<-reports_4 %>%
  mutate(million_reads=(total_reads/1000000)) %>%
  mutate(norm_bacterial_mass=(((sum_bact_count*25)/total_ercc_reads))/million_reads) %>% mutate(norm_bacterial_mass2=(((25*sum_bact_count)/total_ercc_reads))/million_reads)
reports_6<-reports_5 %>% mutate(arm = case_when(arm=="SOC" ~"Standard of care", arm=="DP"~"Doxy-PEP")) %>% 
  mutate(arm = factor(arm,levels=c("Standard of care","Doxy-PEP"))) %>% 
  mutate(visitcode = factor(visitcode,levels=c("Day 0","Month 6"))) %>%
  select(sample_name, visitcode, arm, norm_bacterial_mass2)

  
### stats ###
stat.test_byvisit <-reports_6 %>% 
  mutate(visitcode=as.character(visitcode)) %>% 
  group_by(visitcode) %>% 
  wilcox_test(data=.,norm_bacterial_mass2 ~ arm) %>% 
  add_xy_position(x = "visitcode", dodge=0.8) %>% 
  mutate(y.position = 6)

stat.test_byarm <-reports_6 %>% 
  mutate(visitcode=as.character(visitcode)) %>%
  group_by(arm) %>% 
  wilcox_test(data=.,norm_bacterial_mass2 ~ visitcode) %>% 
  add_xy_position(x = "arm", dodge=0.4) %>%
  mutate(xmin = case_when(arm=="Standard of care" ~ 0.8,arm=="Doxy-PEP" ~ 1.2), 
         xmax = case_when(arm=="Standard of care" ~ 1.8, arm=="Doxy-PEP" ~ 2.2),
         y.position = case_when(arm=="Standard of care" ~ 7,arm=="Doxy-PEP" ~ 8))

stat.test <- stat.test_byvisit %>% 
  bind_rows(stat.test_byarm) %>%
  adjust_pvalue(method = "fdr") %>%
  add_significance("p.adj") %>%
  mutate(p_char = case_when(p.adj <0.1 ~ paste("p.adj =",format(p.adj, scientific = TRUE, digits = 2)),
                            TRUE ~ paste("p.adj =",
        as.character(format(round(p.adj,digits=2), nsmall =2)))),
        y.position=case_when(!is.na(visitcode) ~ y.position-2,
                             arm=="Standard of care" & is.na(visitcode) ~ y.position-2.5,
                             TRUE ~ y.position-3))


### PUBLICATION FIGURE ###

p<-ggplot(reports_6, aes(x=as.character(visitcode), y=norm_bacterial_mass2, color=arm)) +
  scale_color_manual(values = c("#440154FF","#29AF7FFF"), na.value="gray") +
  geom_point(position = position_jitterdodge(0.3,0.2,0.6), alpha=0.6, size=4) +
  geom_boxplot(alpha=0.5, outlier.shape=NA, fill=NA, width=0.6) + 
  xlab("") + ylab("Normalized bacterial mass (pg)\n") + labs(color="") +
  stat_pvalue_manual(stat.test,  label = "{p_char}", tip.length = 0, size=4)  +
  scale_y_log10(labels = trans_format("log10", math_format(10^.x)),
                limits = c(1e-1,1e6)) +
  theme(legend.position = "top")
p


```

## Supplemental Figure 5C: Alpha diversity

``` {r alpha div}

### wrangle data ###
alpha_div<-reports_filterbg_padj %>%
  filter(nt_count>=10, nr_count>=1, nt_alignment_length >= 50) %>%
  filter(p_adj<0.1) %>%
  pivot_wider(names_from = name, values_from = nt_rpm, values_fill = 0, id_cols = sample_name) %>%
  as.data.frame()
rownames(alpha_div)<-toupper(alpha_div$sample_name)
input<-alpha_div[,-1]


### calc alpha div ###
shannon<-diversity(input,index="shannon") %>% as.data.frame()
shan<-shannon %>% mutate(sample_name= rownames(shannon)) %>% rename(shannon_index='.')
simp<-diversity(input,index = "simpson") %>% as.data.frame()
simp2<-simp %>% mutate(sample_name= rownames(simp)) %>% rename(simpson='.')
inv<-diversity(input,index = "invsimpson") %>% as.data.frame()
inv2<-inv %>% mutate(sample_name= rownames(inv)) %>% rename(inv_simpson='.')

### attach onto metadata ###
data1<-left_join(metadata,shan,by="sample_name")
data2<-left_join(data1,simp2,by="sample_name")
data3<-left_join(data2,inv2,by="sample_name")
data4 <-data3 %>% filter(!grepl("NTC|NEC|EMPTY",sample_name)) %>% 
  mutate(arm = case_when(arm=="SOC" ~"Standard of care", arm=="DP" ~ "Doxy-PEP")) %>%
  mutate(arm = factor(arm,levels=c("Standard of care","Doxy-PEP"))) %>%
  mutate(visitcode = factor(visitcode,levels=c("Day 0","Month 6"))) %>%
  filter(sample_name %in% reports_6$sample_name)

### stats ###
stat.test_byvisit_shannon <-data4 %>% 
  mutate(visitcode=as.character(visitcode)) %>% 
  group_by(visitcode) %>% 
  wilcox_test(data=.,shannon_index ~ arm) %>%
  add_xy_position(x = "visitcode", dodge=0.8) %>% 
  mutate(y.position = 5)

stat.test_byarm_shannon <-data4 %>% 
  mutate(visitcode=as.character(visitcode)) %>% 
  group_by(arm) %>% 
  wilcox_test(data=.,shannon_index ~ visitcode) %>% 
  add_xy_position(x = "arm", dodge=0.4) %>%   
  mutate(xmin = case_when(arm=="Standard of care" ~ 0.8,arm=="Doxy-PEP" ~ 1.2), 
         xmax = case_when(arm=="Standard of care" ~ 1.8, arm=="Doxy-PEP" ~ 2.2),
         y.position = case_when(arm=="Standard of care" ~ 6,arm=="Doxy-PEP" ~ 7))

stat.test <- stat.test_byvisit_shannon %>% 
  bind_rows(stat.test_byarm_shannon) %>%
  adjust_pvalue(method = "fdr") %>%
  add_significance("p.adj") %>%
  mutate(p_char = case_when(p.adj<0.01 ~ paste("p.adj =",format(p.adj, scientific = TRUE, digits = 2)),
                            TRUE ~ paste("p.adj =",as.character(format(round(p.adj,digits=2), nsmall = 2)))))


p<-ggplot(data4, aes(x=as.character(visitcode), y=shannon_index, color=arm)) +
  scale_color_manual(values = c("#440154FF","#29AF7FFF"), na.value="gray") +
  geom_point(position = position_jitterdodge(0.3,0.2,0.6), alpha=0.6, size=4) +
  geom_boxplot(alpha=0.5, outlier.shape=NA, fill=NA, width=0.6) + 
  xlab("") + ylab("Shannon Diversity Index\n") + labs(color="") +
  stat_pvalue_manual(stat.test,  label = "{p_char}", tip.length = 0, size=4)  +
  scale_y_continuous(expand=c(0,0), limits = c(0,8)) + 
  theme(legend.position = "top")
p


```

## Supplemental Figure 5E: Beta diversity between DP and SOC arms at Month 6

```{r beta div month 6 DP vs SOC}
### Bacteria ###
my_data<-reports_filterbg_padj %>%
  filter(nt_count>=10, nr_count>=1, nt_alignment_length >= 50) %>%
  filter(p_adj<0.1) %>%
  filter(category=="bacteria") %>%
  dplyr::filter(!grepl("NTC|NEC|EMPTY",sample_name)) %>%
  mutate(sample_name=toupper(sample_name))

combined_data<-left_join(my_data, metadata, by="sample_name")
pruned_data<- combined_data %>%
  filter(visitcode=="Month 6") %>%
  select(ptid, name, nt_rpm) %>%
  pivot_wider(names_from = name, values_from = nt_rpm) %>%
  mutate_if(is.numeric, ~replace_na(., 0)) 

pruned_data2<- pruned_data[,-1]
my_data_df<-as.data.frame(pruned_data2)
rownames(my_data_df) <- pruned_data$ptid
data_transformed<-my_data_df %>% log1p() # log transform the values, with adding 1-all

pruned_metadata <- combined_data %>% select(ptid,arm) %>% distinct() %>% filter(ptid %in% rownames(my_data_df))
beta_dist <- vegdist(my_data_df)
AMR.dispersion<-betadisper(beta_dist, group=pruned_metadata$arm)
AMR.permutest<-permutest(AMR.dispersion)
plot(AMR.dispersion, hull=FALSE, ellipse=TRUE)
set.seed(100)
AMR.div<-adonis2(beta_dist ~ pruned_metadata$arm, data=pruned_metadata, permutations = 1000, method="bray")
print(AMR.div)
MDS<-metaMDS(beta_dist, distance = "bray", k=3, trymax=35, autotransform = TRUE) ##k is the number of dimensions
MDS ##metaMDS takes eaither a distance matrix or your community matrix (then requires method for ‘distance=’)

NMDS1 <- MDS$points[,1] ##also found using scores(birdMDS)
NMDS2 <- MDS$points[,2]
plot<-cbind(pruned_metadata, NMDS1, NMDS2)
cent <- aggregate(cbind(NMDS1, NMDS2) ~ arm, data = pruned_metadata, FUN = mean)
segs<-merge(plot,setNames(cent, c("arm", "oNMDS1", "oNMDS2")), by = "arm", sort = FALSE)
#plot ordination
NMDS_int<-ggplot(plot %>% mutate(arm=factor(arm, levels=c("SOC","DP"))),aes(NMDS1, NMDS2, color=arm)) +
  geom_point() +
  geom_segment(data = segs, mapping = aes(xend = oNMDS1, yend = oNMDS2, alpha=0.5)) + # spiders
  geom_point(data = cent, size = 3) +                         # centroids
  scale_color_manual(labels=c("SOC","DP"), values=c("#440154FF","#29AF7FFF")) +
  scale_fill_manual(labels=c("SOC","DP"), values=c("#440154FF","#29AF7FFF")) +
  annotate("text", x=-1, y=-0.1, label="Standard of Care", hjust = 1, color=c("#440154FF"), size=6) +
  annotate("text", x=0.75, y=0.8, label="Doxy-PEP", hjust = 1, color=c("#29AF7FFF"), size=6) +
  annotate("text", x=-0.5, y=1.4, label=paste("PERMANOVA p-value 0.55"), hjust = 0, size=6) +
                theme(legend.position="none")
NMDS_int




```


## Supplemental Figure 6: Pathogens of interest in DP arm

```{r STI}
### wrangle data ###
data<-reports_filterbg_species_padj %>%
  dplyr::filter(!grepl("NTC|NEC|EMPTY",sample_name)) %>%
  filter(nt_count>=10, nr_count>=1, nt_alignment_length >= 50) %>%
  filter(p_adj<0.1) %>%
  pivot_wider(names_from = name, values_from = nt_rpm, values_fill = 0, id_cols = sample_name) %>%
  as.data.frame()
trimmed<-data %>% select(sample_name,`Chlamydia trachomatis`,`Neisseria gonorrhoeae`, `Mycoplasma genitalium`, `Clostridioides difficile`) %>%
  mutate(sample_name=toupper(sample_name))

trimmed_2<-left_join(trimmed,metadata,by="sample_name")

trimmed_3<-trimmed_2 %>% mutate(arm = case_when(arm=="SOC" ~"Standard of care", arm=="DP" ~ "Doxy-PEP")) %>% mutate(arm = factor(arm,levels=c("Standard of care","Doxy-PEP"))) %>% mutate(`Neisseria gonorrhoeae`=`Neisseria gonorrhoeae`+1) %>% mutate(visitcode = factor(visitcode,levels=c("Day 0","Month 6"))) %>% mutate(`Neisseria gonorrhoeae`=`Neisseria gonorrhoeae`+1) %>%
  mutate(`Chlamydia trachomatis`=`Chlamydia trachomatis`+1) %>% mutate(`Mycoplasma genitalium`=`Mycoplasma genitalium`+1)


# N. gonorrhoeae
stat.test_onlydoxy_ng <-trimmed_3 %>% 
  filter(arm=="Doxy-PEP") %>% 
  mutate(visitcode=as.character(visitcode)) %>%
  wilcox_test(data=.,`Neisseria gonorrhoeae` ~ visitcode) %>% 
  mutate(y.position = 3.5, p_char = case_when(p<0.1 ~ paste("p =",format(p, scientific = TRUE, digits = 2)), TRUE ~ paste("p =", as.character(format(round(p, digits=2), nsmall =2)))))

trimmed_4<-trimmed_3 %>% filter(arm=="Doxy-PEP")  
p<-ggplot(trimmed_4, aes(x=as.character(visitcode), y=`Neisseria gonorrhoeae`, color=arm))  + geom_jitter(alpha=0.6, size=4, position=position_jitter(width=0.1, height=0.1)) + geom_boxplot(alpha=0.5, outlier.shape=NA, fill=NA, width=0.6) + scale_y_log10() + 
    scale_color_manual(values = c("#29AF7FFF"), na.value="gray") +
  xlab("") + ylab("Neisseria gonorrhoeae log10(NT RPM)") + labs(color="") +
  stat_pvalue_manual(stat.test_onlydoxy_ng,  label = "{p_char}", tip.length = 0, size=5)
p

# Chlamydia
stat.test_onlydoxy_ct <-trimmed_3 %>% 
  filter(arm=="Doxy-PEP") %>% 
  mutate(visitcode=as.character(visitcode)) %>%
  wilcox_test(data=.,`Chlamydia trachomatis` ~ visitcode) %>% 
  mutate(y.position = 1, p_char = case_when(p<0.1 ~ paste("p =",format(p, scientific = TRUE, digits = 2)), TRUE ~ paste("p =", as.character(format(round(p, digits=2), nsmall =2)))))  

trimmed_4<-trimmed_3 %>% filter(arm=="Doxy-PEP")  
p<-ggplot(trimmed_4, aes(x=as.character(visitcode), y=`Chlamydia trachomatis`, color=arm))  + geom_jitter(alpha=0.6, size=4, position=position_jitter(width=0.1, height=0.1)) + geom_boxplot(alpha=0.5, outlier.shape=NA, fill=NA, width=0.6) + scale_y_log10() + 
    scale_color_manual(values = c("#29AF7FFF"), na.value="gray") +
  xlab("") + ylab("Chlamydia trachomatis log10(NT RPM)") + labs(color="") +
  stat_pvalue_manual(stat.test_onlydoxy_ct,  label = "{p_char}", tip.length = 0, size=5)
p

# M. genitalium
stat.test_onlydoxy_mg <-trimmed_3 %>% 
  filter(arm=="Doxy-PEP") %>% 
  mutate(visitcode=as.character(visitcode)) %>%
  wilcox_test(data=.,`Mycoplasma genitalium` ~ visitcode) %>% 
  mutate(y.position = 1.75, p_char = case_when(p<0.1 ~ paste("p =",format(p, scientific = TRUE, digits = 2)), TRUE ~ paste("p =", as.character(format(round(p, digits=2), nsmall =2)))))  

trimmed_4<-trimmed_3 %>% filter(arm=="Doxy-PEP")  
p<-ggplot(trimmed_4, aes(x=as.character(visitcode), y=`Mycoplasma genitalium`, color=arm))  + geom_jitter(alpha=0.6, size=4, position=position_jitter(width=0.1, height=0.1)) + geom_boxplot(alpha=0.5, outlier.shape=NA, fill=NA, width=0.6) + scale_y_log10() + 
    scale_color_manual(values = c("#29AF7FFF"), na.value="gray") +
  xlab("") + ylab("Mycoplasma genitalium log10(NT RPM)") + labs(color="") +
  stat_pvalue_manual(stat.test_onlydoxy_mg,  label = "{p_char}", tip.length = 0, size=5)
p

# C. diff
stat.test_onlydoxy_cd <-trimmed_3 %>% 
  filter(arm=="Doxy-PEP") %>% 
  mutate(visitcode=as.character(visitcode)) %>%
  wilcox_test(data=.,`Clostridioides difficile` ~ visitcode) %>% 
  mutate(y.position = 4, p_char = case_when(p<0.1 ~ paste("p =",format(p, scientific = TRUE, digits = 2)), TRUE ~ paste("p =", as.character(format(round(p, digits=2), nsmall =2)))))  

trimmed_4<-trimmed_3 %>% filter(arm=="Doxy-PEP")  
p<-ggplot(trimmed_4, aes(x=as.character(visitcode), y=`Clostridioides difficile`, color=arm))  + geom_jitter(alpha=0.6, size=4, position=position_jitter(width=0.1, height=0.1)) + geom_boxplot(alpha=0.5, outlier.shape=NA, fill=NA, width=0.6) + scale_y_log10() + 
    scale_color_manual(values = c("#29AF7FFF"), na.value="gray") +
  xlab("") + ylab("Clostridioides difficile log10(NT RPM)") + labs(color="") +
  stat_pvalue_manual(stat.test_onlydoxy_cd,  label = "{p_char}", tip.length = 0, size=5)
p

```


# --------------------
# RESULTS: DIFFERENTIAL ABUNDANCE

## Differential abundance over time in DP arm

```{r deseq}

dds = DESeq(ddsdp)
resDEdp=results(dds,cooksCutoff=FALSE, contrast=c("visitcode","6","0"))

# order the results by log fold change
resDEdp=resDEdp[order(resDEdp$log2FoldChange,decreasing = T),]

#thresholds
min_lfc = 1 
max_padj = 0.05
min_basemean = 250

resDEdp_filtered = resDEdp[resDEdp$baseMean > min_basemean,]


DEframedp = data.frame(rownames(resDEdp_filtered))
DEframedp = cbind(DEframedp, resDEdp_filtered$log2FoldChange,resDEdp_filtered$padj)
colnames(DEframedp) = c("taxon","log2FoldChange","padj")


DEframedp$diffexpressed = "NoChange"
DEframedp$diffexpressed[DEframedp$log2FoldChange > min_lfc & DEframedp$padj < max_padj] = "Up"
DEframedp$diffexpressed[DEframedp$log2FoldChange < -min_lfc & DEframedp$padj < max_padj] = "Down"
DEframedp$delabel = NA
DEframedp<-DEframedp %>% mutate(delabel= case_when(diffexpressed!="NoChange"~ taxon))

doxypep_de <- 
ggplot(data=DEframedp, aes(x=log2FoldChange, y=-log10(padj), col=diffexpressed, label=delabel)) +
        geom_point() + 
        theme_minimal() +
        geom_text_repel(size=3) +
        scale_color_manual(values=c("blue", "black", "red")) +
        geom_vline(xintercept=c(-min_lfc, min_lfc), col="black") +
        geom_hline(yintercept=-log10(max_padj), col="black")

DEframedp %>% filter(padj<=0.05)

```


```{r metagenomeseq}

da_mgs_md <- metadata %>%
  filter(sample_name %in% reports_6$sample_name) %>%
  filter(arm=="DP")
rownames(da_mgs_md) <- da_mgs_md$sample_name

da_mgsdf <- reports_filterbg_padj %>% select(sample_name, genus_tax_id, nt_count) %>%
  mutate(sample_name=toupper(sample_name)) %>%
  filter(sample_name %in% da_mgs_md$sample_name) %>%
  group_by(sample_name, genus_tax_id) %>% summarise(count=sum(nt_count)) %>% ungroup() %>%
  pivot_wider(names_from = sample_name, values_from = count, values_fill = list(count = 0)) %>%
  column_to_rownames("genus_tax_id")


obj <- newMRexperiment(da_mgsdf)
p <- cumNormStatFast(obj)
obj <- cumNorm(obj, p = p)
res <- fitFeatureModel(obj, model.matrix(~ visitcode, data = da_mgs_md))

# Extract results
results <- MRfulltable(res)
results.adj <- MRfulltable(res) %>% filter(adjPvalues<=0.05)

# no significant bacterial genera after adjustment using BH

```


```{r ALDEx2}

# Prepare the data for ALDEx2

aldex2_md <- metadata %>%
  filter(sample_name %in% reports_6$sample_name) %>%
  filter(arm=="DP")
rownames(aldex2_md) <- aldex2_md$sample_name

aldex2_mgsdf <- reports_filterbg_padj %>% select(sample_name, genus_tax_id, nt_count) %>%
  mutate(sample_name=toupper(sample_name)) %>%
  filter(sample_name %in% aldex2_md$sample_name) %>%
  group_by(sample_name, genus_tax_id) %>% summarise(count=sum(nt_count)) %>% ungroup() %>%
  pivot_wider(names_from = sample_name, values_from = count, values_fill = list(count = 0)) %>%
  column_to_rownames("genus_tax_id")

conds <- aldex2_md$visitcode

# Perform CLR transformation and Monte Carlo sampling
aldex_data <- aldex.clr(aldex2_mgsdf, conds, mc.samples = 128, denom = "all")

# Perform differential abundance testing
aldex_results <- aldex.ttest(aldex_data)

# Add effect size and summary statistics
aldex_effect <- aldex.effect(aldex_data)
aldex_summary <- cbind(aldex_results, aldex_effect)

aldex_summary_adj <- aldex_summary %>% filter(we.eBH<=0.05)

# no significant bacterial genera after adjustment using BH

```
